src/library/Checkbox/Checkbox.js

```diff
@@ -127,4 +191,46 @@ export default class Checkbox extends PureComponent<Props> {
 
     return <Root {...rootProps} />;
   }
+
+  setInputRef = (node: HTMLInputElement) => {
+    const { inputRef } = this.props;
+    this.input = node;
+    inputRef && inputRef(node);
+  };
+
+  updateState = (indeterminate: boolean, checked: boolean) => {
```

Comment:
I hear you; valid concern.  It is still state though, just not React state.  Also, since it is updating values on both the component instance and the input, I think I'll leave it as is.